packages/pluggableWidgets/popup-menu-web/src/PopupMenu.tsx

```diff
@@ -0,0 +1,141 @@
+// import { isBehindRandomElement, unBlockElement } from "@mendix/pluggable-widgets-tools";
+import {
+    isBehindElement,
+    isBehindRandomElement,
+    isElementPartiallyOffScreen,
+    isElementVisibleByUser,
+    moveAbsoluteElementOnScreen,
+    unBlockAbsoluteElement
+} from "../../../tools/pluggable-widgets-tools/src/index";
+import { ReactElement, useState, createElement, useCallback, RefObject, useEffect, useRef } from "react";
+import { executeAction } from "@widgets-resources/piw-utils";
+import { ActionValue } from "mendix";
+
+import { PopupMenuContainerProps } from "../typings/PopupMenuProps";
+import "./ui/PopupMenu.scss";
+
+export default function PopupMenu(props: PopupMenuContainerProps): ReactElement {
+    const ref = useRef<HTMLDivElement>(null);
+    useOnClickOutside(ref, () => setVisibility(false));
+    const [visibility, setVisibility] = useState(false);
+    useEffect(() => {
+        const element = (ref.current as HTMLDivElement).querySelector(".popupmenu-menu") as HTMLDivElement;
```

Comment:
Howcome you cast here? What about instead handling the `null`-able state of `current` with the elvis operator, then guarding subsequent statements? `ref.current?.querySelector()`

Reply:
Nice one, thanks!

Reply:
Considering your subsequent change, I believe we may run into a TypeError. The expression is cast to `HTMLDivElement` even though you used the elvis operator. Essentially you are swallowing the possible `null` state, which means that TS thinks element is always a div element. 

If you cast to `HTMLDivElement` | null then TS should suggest guards for subsuequent code in the block.

Reply:
Addressed.